_dev/src/ts/pages/UpdatePageUpdateOptions.ts

```diff
@@ -1,13 +1,51 @@
 import UpdatePage from './UpdatePage';
+import api from '../api/RequestHandler';
 
 export default class UpdatePageUpdateOptions extends UpdatePage {
   protected stepCode = 'update-options';
 
-  constructor() {
-    super();
-  }
-
   public mount() {
     this.initStepper();
+    this.form.addEventListener('submit', this.onSubmit);
+    this.form.addEventListener('change', this.onChange);
+  }
+
+  public beforeDestroy() {
+    this.form.removeEventListener('submit', this.onSubmit);
+    this.form.removeEventListener('change', this.onChange);
+  }
+
+  private get form(): HTMLFormElement {
+    const form = document.forms.namedItem('update-options-page-form');
+    if (!form) {
+      throw new Error('Form not found');
+    }
+
+    ['routeToSave', 'routeToSubmit'].forEach((data) => {
+      if (!form.dataset[data]) {
+        throw new Error(`Missing data ${data} from form dataset.`);
+      }
+    });
+
+    return form;
   }
+
+  private readonly onChange = async (ev: Event) => {
+    const optionInput = ev.target as HTMLInputElement;
+
+    optionInput.setAttribute('disabled', 'true');
+
+    const data = new FormData();
+    data.append('name', optionInput.name);
+    data.append('value', JSON.stringify(optionInput.checked));
+    await api.post(this.form.dataset.routeToSave!, data);
```

Comment:
@ga-devfront I've kept this implementation after trying switching on the whole form in FormData. Unfortunately the options are only sent if they are checks, which does not fit the ConfigurationValidator behavior where only updated data is expected.

Reply:
you can do it like this to get all the form with all chebox true or false =>
```
const checkboxStates = Array.from(this.form.elements)
  .filter(el => el.type === 'checkbox')
  .reduce((acc, checkbox) => {
    acc[checkbox.name] = checkbox.checked;
    return acc;
  }, {});
  ```